lib/rubocop/cop/rspec/pending_without_reason.rb

```diff
@@ -116,8 +116,9 @@ def skipped_without_reason?(node)
             skipped_by_metadata_without_reason?(node)
         end
 
-        def inside_example?(node)
-          node.ancestors.find { |parent| example?(parent) }
+        def directly_inside_example?(node)
+          example?(node.parent) ||
+            (node.parent.if_type? && example?(node.parent.parent.parent))
```

Comment:
```
> node.parent.parent.parent
=> s(:block,
  s(:send, nil, :it,
    s(:str, "does something")),
  s(:args),
  s(:begin,
    s(:if,
      s(:send,
        s(:const, nil, :RUBY_VERSION), :<,
        s(:str, "3.0")),
      s(:send, nil, :pending), nil),
    s(:if,
      s(:send,
        s(:const, nil, :RUBY_VERSION), :<,
        s(:str, "3.0")),
      s(:send, nil, :skip), nil)))
```

I believe `begin` will always be there for a multi-statement block. Unless the real meat of the test is inside the `else`:
```ruby
it do
  if RUBY_VERSION < '3.0'
    skip
  else
    expect(...).to ...
  end
end
```
But it looks quite weird and doesn't make much sense.

Reply:
I tried to take on a local variable, but I am a little unsure if this variable name is optimal.